src/Components/src/Microsoft.AspNetCore.Components/Rendering/Renderer.cs

```diff
@@ -295,5 +293,35 @@ private void RemoveEventHandlerIds(ArrayRange<int> eventHandlerIds, Task afterTa
                     RemoveEventHandlerIds(eventHandlerIdsClone, Task.CompletedTask));
             }
         }
+
+        /// <summary>
+        /// Releases all resources currently used by this <see cref="Renderer"/> instance.
+        /// </summary>
+        /// <param name="disposing"><see langword="true"/> if this method is being invoked by <see cref="IDisposable.Dispose"/>, otherwise <see langword="false"/>.</param>
+        protected virtual void Dispose(bool disposing)
+        {
+            foreach (var componentState in _componentStateById.Values)
+            {
+                if (componentState.Component is IDisposable disposable)
+                {
+                    try
+                    {
+                        disposable.Dispose();
+                    }
+                    catch
+                    {
+                        // Ignore failures in components throwing
```

Comment:
Although I agree that we shouldn't stop the `foreach` if there's an exception, I'm not convinced we should be swallowing the exceptions altogether. Do you think it would be reasonable to capture them all in an `AggregateException` and then throw it at the end of this method?